lms/js_config_types.py

```diff
@@ -37,13 +44,23 @@ class APIAssignment(TypedDict):
     stats: NotRequired[AssignmentStats]
 
 
+class DashboardLinks(TypedDict):
```

Comment:
So these are the URLs that will usually appear in the browser's navigation bar, while the ones under `routes` are the ones meant to be consumed as JSON async calls, correct?

There's another option to avoid this duplication, which is letting the FE handle them, as it seems we are finally going to be handling navigation client-side.

If the server can define a route like `/dashboard/*`, where * can be anything, we can make it match the dashboard mini-app view (not sure if "mini" is accurate anymore 😅), and then the FE routing logic will do the proper view rendering based on that.

If I'm not wrong, the BE is already serving the same "empty" template for all dashboard routes, so we may as well centralize that.

I may be missing something though. What do you think.

Reply:
The BE needs to know about these for when the user arrives directly to them. We serve the mini app on those.

I reckon, but I'm not 100% sure, that the FE will use the same ones while navigating between "dashboards" even if is doing client side navigation the URL will be the same.

I didn't realized that the FE will also need to match on these to decide in which "mode" they are, basing that matching on this templates is probably tricky so that might be the duplication we'll have to live with.

Reply:
> If the server can define a route like /dashboard/*, where * can be anything, we can make it match the dashboard mini-app view (not sure if "mini" is accurate anymore 😅), and then the FE routing logic will do the proper view rendering based on that.

Ah, reread this and understood it more now.

The BE routes are now different because they do two things:

- They check for the permission on the "entity"
- The display the title of the entity.

I don't think there's a good reason to do a check on permissions if we are not exposing anything about the object itself. The API will do the check.

The only thing we expose now is the title so we could stop setting the title in the BE, would that be ugly while it loads?